patches/api/0333-Added-getHostname-to-AsyncPlayerPreLoginEvent.patch

```diff
@@ -0,0 +1,59 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: MCMDEV <john-m.1@gmx.de>
+Date: Fri, 24 Sep 2021 17:59:23 +0200
+Subject: [PATCH] Added getHostname to AsyncPlayerPreLoginEvent
+
+
+diff --git a/src/main/java/org/bukkit/event/player/AsyncPlayerPreLoginEvent.java b/src/main/java/org/bukkit/event/player/AsyncPlayerPreLoginEvent.java
+index 694a81769076ea58aae9f14f076ab80c9952c957..ac880f5442e880717079acba10e98d3ef6c489f6 100644
+--- a/src/main/java/org/bukkit/event/player/AsyncPlayerPreLoginEvent.java
++++ b/src/main/java/org/bukkit/event/player/AsyncPlayerPreLoginEvent.java
+@@ -22,6 +22,7 @@ public class AsyncPlayerPreLoginEvent extends Event {
+     private final InetAddress ipAddress;
+     private final InetAddress rawAddress; // Paper
+     //private UUID uniqueId; // Paper - Not used anymore
++    private final String hostname; // Paper
+ 
+     @Deprecated
+     public AsyncPlayerPreLoginEvent(@NotNull final String name, @NotNull final InetAddress ipAddress) {
+@@ -63,10 +64,10 @@ public class AsyncPlayerPreLoginEvent extends Event {
+ 
+     @Deprecated
+     public AsyncPlayerPreLoginEvent(@NotNull final String name, @NotNull final InetAddress ipAddress, @NotNull final UUID uniqueId, @NotNull PlayerProfile profile) {
+-        this(name, ipAddress, ipAddress, uniqueId, profile);
++        this(name, ipAddress, ipAddress, uniqueId, profile, "");
+     }
+ 
+-    public AsyncPlayerPreLoginEvent(@NotNull final String name, @NotNull final InetAddress ipAddress, @NotNull final InetAddress rawAddress, @NotNull final UUID uniqueId, @NotNull PlayerProfile profile) {
++    public AsyncPlayerPreLoginEvent(@NotNull final String name, @NotNull final InetAddress ipAddress, @NotNull final InetAddress rawAddress, @NotNull final UUID uniqueId, @NotNull PlayerProfile profile, @NotNull String hostname) {
```

Comment:
Since this is "technically" API, you can't break it by adding a new param to this ctor, you have to make a new ctor with the added param and deprecate the now-old one.

Reply:
This should be fixed now